src/typeahead/highlight.ts

```diff
@@ -30,28 +30,30 @@ export class NgbHighlight implements OnChanges {
    * The text highlighting is added to.
    *
    * If the `term` is found inside this text, it will be highlighted.
+   * If the `term` contains array then all the items from it will be highlighted inside the text.
    */
-  @Input() result: string;
+  @Input() result: any;
 
   /**
-   * The term to be highlighted.
+   * The term or array of terms to be highlighted.
    */
-  @Input() term: string;
+  @Input() term: any;
 
   ngOnChanges(changes: SimpleChanges) {
     const resultStr = toString(this.result);
-    const resultLC = resultStr.toLowerCase();
-    const termLC = toString(this.term).toLowerCase();
-    let currentIdx = 0;
+    if (!resultStr) {
+      this.parts = [this.result];
+      return;
+    }
+    let resultTerms: string[] = Array.isArray(this.term) ? this.term.map(x => toString(x)) : [toString(this.term)];
```

Comment:
Could you please also check the previous PR and my comment → https://github.com/ng-bootstrap/ng-bootstrap/pull/2750/files

I think combining the previous one with yours could make the code simpler ?

Reply:
thanks @maxokorokov ! will check the previous attempt and get back with best wisdom from both prs.